metron-platform/metron-hbase/metron-hbase-common/src/test/java/org/apache/metron/hbase/mock/MockHTable.java

```diff
@@ -44,651 +35,757 @@
-            break;
-          }
-          // ignoring next key hint which is a optimization to reduce file system IO
+        if (filter.hasFilterRow() && !filteredOnRowKey) {
+            filter.filterRowCells(nkvs);
         }
-        if (filter.hasFilterRow()) {
-          filter.filterRow();
+        if (filter.filterRow() || filteredOnRowKey) {
+            nkvs.clear();
         }
-        kvs = nkvs;
-      }
-      if (!kvs.isEmpty()) {
-        ret.add(new Result(kvs));
-      }
-    }
-
-    return new ResultScanner() {
-      private final Iterator<Result> iterator = ret.iterator();
-      @Override
-      public Iterator<Result> iterator() {
-        return iterator;
-      }
-      @Override
-      public Result[] next(int nbRows) throws IOException {
-        ArrayList<Result> resultSets = new ArrayList<Result>(nbRows);
-        for(int i = 0; i < nbRows; i++) {
-          Result next = next();
-          if (next != null) {
-            resultSets.add(next);
-          } else {
-            break;
-          }
+        tmp = nkvs;
+        return tmp;
+    }
+
+    private boolean check(byte[] row, byte[] family, byte[] qualifier, CompareFilter.CompareOp compareOp, byte[] value) {
+        if (value == null)
+            return !data.containsKey(row) ||
+                    !data.get(row).containsKey(family) ||
+                    !data.get(row).get(family).containsKey(qualifier);
+        else if (data.containsKey(row) &&
+                data.get(row).containsKey(family) &&
+                data.get(row).get(family).containsKey(qualifier) &&
+                !data.get(row).get(family).get(qualifier).isEmpty()) {
+
+            byte[] oldValue = data.get(row).get(family).get(qualifier).lastEntry().getValue();
+            int compareResult = Bytes.compareTo(value, oldValue);
+            switch (compareOp) {
+                case LESS:
+                    return compareResult < 0;
+                case LESS_OR_EQUAL:
+                    return compareResult <= 0;
+                case EQUAL:
+                    return compareResult == 0;
+                case NOT_EQUAL:
+                    return compareResult != 0;
+                case GREATER_OR_EQUAL:
+                    return compareResult >= 0;
+                case GREATER:
+                    return compareResult > 0;
+                default:
+                    throw new RuntimeException("Unknown Compare op " + compareOp.name());
+            }
+        } else {
+            return false;
         }
-        return resultSets.toArray(new Result[resultSets.size()]);
-      }
-      @Override
-      public Result next() throws IOException {
-        try {
-          return iterator().next();
-        } catch (NoSuchElementException e) {
-          return null;
+    }
+
+    @Override
+    public void close() throws IOException {
+
+    }
+
+    @Override
+    public CoprocessorRpcChannel coprocessorService(byte[] row) {
```

Comment:
We don't need this for the integration tests currently and removed them in our original port